admin/src/core/services/cognito.service.ts

```diff
@@ -0,0 +1,214 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { Amplify, Auth } from "aws-amplify";
+import { AwsCognitoConfig } from "@api-client";
+import { User } from "@utility/security/user";
+import { ConfigService } from "@utility/services/config.service";
+import { getFakeUser } from "./mock-user";
+import type { CognitoUserSession } from "amazon-cognito-identity-js";
+
+@Injectable()
+export class CognitoService {
+  private config: AwsCognitoConfig = {
```

Comment:
Not sure why we start with a blank config.

Reply:
Good find!
The aws-amplify config file is expecting a nested structure of objects. Initially, I was thinking the same but I didn't know a way at backend to use nested type class for NestJs to produce the same structure. That's why Catherine at frontend needs to map values to correct object before passing into aws-amplify lib.
I spent a little time and resolved the issue. Code updated and tested. It's good!